subsys/bluetooth/services/ras/rreq/ras_rreq.c

```diff
@@ -0,0 +1,714 @@
+				"waiting for RAS-CP response");
+			return BT_GATT_ITER_CONTINUE;
+		}
+
+		LOG_DBG("Ranging counter %d overwritten whilst receiving", ranging_counter);
+		rreq->on_demand_rd.error_with_data_receive = true;
+		data_receive_finished(rreq);
+
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	rreq->rd_overwritten.cb(conn, ranging_counter);
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int ack_ranging_data(struct bt_ras_rreq *rreq)
+{
+	int err;
+
+	NET_BUF_SIMPLE_DEFINE(ack_ranging_data, RASCP_CMD_OPCODE_LEN + sizeof(uint16_t));
+	net_buf_simple_add_u8(&ack_ranging_data, RASCP_OPCODE_ACK_RD);
+	net_buf_simple_add_le16(&ack_ranging_data, rreq->on_demand_rd.counter_in_progress);
+
+	err = bt_gatt_write_without_response(rreq->conn, rreq->cp.subscribe_params.value_handle,
+					     ack_ranging_data.data, ack_ranging_data.len, false);
+	if (err) {
+		LOG_DBG("CP ACK ranging data written failed, err %d", err);
+		return err;
+	}
+
+	rreq->cp.state = BT_RAS_RREQ_CP_STATE_ACK_RD_WRITTEN;
+	LOG_DBG("Ack Ranging data for counter %d", rreq->on_demand_rd.counter_in_progress);
+
+	return 0;
+}
+
+static void handle_rsp_code(uint8_t rsp_code, struct bt_ras_rreq *rreq)
+{
+	switch (rreq->cp.state) {
+	case BT_RAS_RREQ_CP_STATE_NONE: {
+		if (rreq->on_demand_rd.data_get_in_progress &&
+		    rsp_code == RASCP_RESPONSE_PROCEDURE_NOT_COMPLETED) {
+			LOG_DBG("Ranging counter %d aborted whilst receiving",
+				rreq->on_demand_rd.counter_in_progress);
+			rreq->on_demand_rd.error_with_data_receive = true;
+			data_receive_finished(rreq);
+			break;
+		}
+
+		LOG_DBG("Unexpected Response code received %d", rsp_code);
+		break;
+	}
+	case BT_RAS_RREQ_CP_STATE_GET_RD_WRITTEN: {
+		__ASSERT_NO_MSG(rreq->on_demand_rd.data_get_in_progress);
+		rreq->cp.state = BT_RAS_RREQ_CP_STATE_NONE;
+
+		if (rsp_code != RASCP_RESPONSE_SUCCESS) {
+			LOG_DBG("Get Ranging Data returned an error %d", rsp_code);
+			rreq->on_demand_rd.error_with_data_receive = true;
+			data_receive_finished(rreq);
+			break;
+		}
+
+		LOG_DBG("Get Ranging Data Success");
+
+		break;
+	}
+	case BT_RAS_RREQ_CP_STATE_ACK_RD_WRITTEN: {
+		__ASSERT_NO_MSG(rreq->on_demand_rd.data_get_in_progress);
+		rreq->cp.state = BT_RAS_RREQ_CP_STATE_NONE;
+		if (rsp_code != RASCP_RESPONSE_SUCCESS) {
+			LOG_DBG("ACK Ranging Data returned an error %d", rsp_code);
+		}
+
+		data_receive_finished(rreq);
+		break;
+	}
+	default:
+		LOG_DBG("Unexpected Response code received %d", rsp_code);
+		break;
+	}
+}
+
+static uint8_t ras_cp_notify_func(struct bt_conn *conn, struct bt_gatt_subscribe_params *params,
+				  const void *data, uint16_t length)
+{
+	struct net_buf_simple rsp;
+	struct bt_ras_rreq *rreq = ras_rreq_find(conn);
+
+	net_buf_simple_init_with_data(&rsp, (uint8_t *)data, length);
+
+	uint8_t opcode = net_buf_simple_pull_u8(&rsp);
+
+	switch (opcode) {
+	case RASCP_RSP_OPCODE_COMPLETE_RD_RSP: {
+		if (rsp.len != RASCP_RSP_OPCODE_COMPLETE_RD_RSP_LEN) {
+			LOG_DBG("RAS-CP Complete RD RSP incorrect length: %d", length);
```

Comment:
Same with LOG_DBGs in this function. They look like are for invalid cases. So either WRN or ERR could be better